tasks/sync_repo_languages.py

```diff
@@ -46,45 +45,50 @@ def run_impl(
             and len(desired_languages_intersection) == 0
         ) or manual_trigger
 
+        if not should_sync_languages:
+            return {"successful": True, "synced": False}
+
+        log_extra = dict(
+            owner_id=repository.ownerid or "",
+            repository_id=repository.repoid,
+        )
+        log.info("Syncing repository languages", extra=log_extra)
+
+        installation_name_to_use = get_installation_name_for_owner_for_task(
+            self.name, repository.owner
+        )
+        repository_service = get_repo_provider_service(
+            repository, installation_name_to_use=installation_name_to_use
+        )
```

Comment:
I can remember that I may have created "yet another" wrapper that takes care of this `installation_name_to_use`, and introduces a cache for these things.
Or I tried and failed because these things are not cache-able.